qutip/core/cy/qobjevo.pyx

```diff
@@ -0,0 +1,1114 @@
+        coefficients are applied.
+
+    copy : bool
+        If Q_object is already a QobjEvo, return a copy.
+
+    Attributes
+    ----------
+    dims : list
+        List of dimensions keeping track of the tensor structure.
+
+    shape : (int, int)
+        List of dimensions keeping track of the tensor structure.
+
+    Property
+    --------
+    num_obj
+        Number of parts composing the system.
+
+    const:
+        Does the system change depending on `t`.
+
+    isoper:
+        Indicates if the system represents an operator.
+
+    issuper:
+        Indicates if the system represents an superoperator.
+
+    Methods
+    -------
+    copy() :
+        Create copy of Qobj
+
+    arguments(new_args):
+        Update the args of the object
+
+    Math:
+        +/- QobjEvo, Qobj, scalar:
+            Addition is possible between QobjEvo and with Qobj or scalar
+        -:
+            Negation operator
+        * Qobj, scalar:
+            Product is possible with Qobj or scalar
+        / scalar:
+            It is possible to divide by scalar only
+    conj()
+        Return the conjugate of quantum object.
+
+    dag()
+        Return the adjoint (dagger) of quantum object.
+
+    trans()
+        Return the transpose of quantum object.
+
+    &
+        tensor between Quantum Object
+
+    linear_map(op_mapping)
+        Apply the function f to every Qobj. f(Qobj) -> Qobj
+
+    tidyup(atol=1e-12)
+        Removes small elements from quantum object.
+
+    compress():
+        Merge ops which are based on the same quantum object and coeff type.
+
+    __call__(t, args={}, data=False):
+        Return the Qobj at time t.
+
+    matmul(t, mat):
+        Product of this at t time with a Qobj.
+
+    expect(t, psi, herm=False):
+        Calculates the expectation value for the quantum object (if operator,
+            no check) and state psi.
+        Return only the real part if herm.
+
+    to(data_type):
+        Transform all component to a common data_type.
+
+    """
+    def __init__(QobjEvo self, Q_object=[], args={}, tlist=None, copy=True):
+        if isinstance(Q_object, QobjEvo):
+            self.dims = Q_object.dims
+            self.shape = Q_object.shape
+            self._shift_dt = (<QobjEvo> Q_object)._shift_dt
+            self.func_qobj = {}
+            if copy:
+                self.args = (<QobjEvo> Q_object).args.copy()
+                self.elements = [element.copy() for element in (<QobjEvo> Q_object).elements]
+                self.funcQEvo = (<QobjEvo> Q_object).funcQEvo.copy()
+            else:
+                self.args = (<QobjEvo> Q_object).args
+                self.elements = (<QobjEvo> Q_object).elements
+                self.funcQEvo = (<QobjEvo> Q_object).funcQEvo
+            if args:
+                self.arguments(args)
+            return
+
+        self.elements = []
+        self.funcQEvo = set()
+        self.func_qobj = {}
```

Comment:
When I added `self.func_qobj` it made `self.funcQEvo` redundant.
It is a dict of `{func: funct(t)}`so that the like of `op * op.dag()` only call each function once.

Reply:
Ok cool, that seems sensible.